firebase-crashlytics/src/androidTest/java/com/google/firebase/crashlytics/internal/common/FirebaseCrashlyticsReportManagerTest.java

```diff
@@ -0,0 +1,156 @@
+    final CrashlyticsReport mockReport = mock(CrashlyticsReport.class);
+    when(dataCapture.captureReportData(anyString(), anyLong())).thenReturn(mockReport);
+
+    reportManager.onBeginSession(sessionId);
+
+    verify(dataCapture).captureReportData(sessionId, timestampSeconds);
+    verify(reportPersistence).persistReport(mockReport);
+  }
+
+  @Test
+  public void testOnFatalEvent_persistsFatalEventForSessionId() {
+    final String eventType = "crash";
+    final Exception exceptionEvent = new Exception("fatal");
+    final Thread eventThread = Thread.currentThread();
+
+    final String sessionId = "testSessionId";
+    final long timestamp = System.currentTimeMillis();
+    when(mockCurrentTimeProvider.getCurrentTimeMillis()).thenReturn(timestamp);
+    final long timestampSeconds = timestamp / 1000;
+    final CrashlyticsReport.Session.Event mockEvent = mock(CrashlyticsReport.Session.Event.class);
+    when(dataCapture.captureEventData(
+            any(Throwable.class), any(Thread.class), anyString(), anyLong(), anyInt(), anyInt()))
+        .thenReturn(mockEvent);
+
+    reportManager.onBeginSession(sessionId);
+    reportManager.onFatalEvent(exceptionEvent, eventThread);
+
+    verify(dataCapture)
+        .captureEventData(exceptionEvent, eventThread, eventType, timestampSeconds, 4, 8);
+    verify(reportPersistence).persistEvent(mockEvent, sessionId);
+  }
+
+  @Test
+  public void onSessionsFinalize_finalizesReports() {
+    final String sessionId = "testSessionId";
+    reportManager.onBeginSession(sessionId);
+    reportManager.onFinalizeSessions();
+
+    verify(reportPersistence).finalizeReports(sessionId);
+  }
+
+  @Test
+  @SuppressWarnings("unchecked")
+  public void onReportSend_successfulReportsAreDeleted() throws InterruptedException {
+    final String orgId = "testOrgId";
+    final String sessionId1 = "sessionId1";
+    final String sessionId2 = "sessionId2";
+
+    final AppSettingsData appSettings =
+        new AppSettingsData(null, null, null, null, null, orgId, false);
+
+    final List<CrashlyticsReport> finalizedReports = new ArrayList<>();
+    final CrashlyticsReport mockReport1 = mockReport(sessionId1, orgId);
+    final CrashlyticsReport mockReport2 = mockReport(sessionId2, orgId);
+    finalizedReports.add(mockReport1);
+    finalizedReports.add(mockReport2);
+
+    when(reportPersistence.loadFinalizedReports()).thenReturn(finalizedReports);
+
+    final Task<CrashlyticsReport> successfulTask = Tasks.forResult(mockReport1);
+    final Task<CrashlyticsReport> failedTask = Tasks.forException(new Exception("fail"));
+
+    when(reportSender.sendReport(mockReport1)).thenReturn(successfulTask);
+    when(reportSender.sendReport(mockReport2)).thenReturn(failedTask);
+
+    reportManager.onSendReports(appSettings);
+
+    verify(reportSender).sendReport(mockReport1);
+    verify(reportSender).sendReport(mockReport2);
+
+    try {
+      Tasks.await(successfulTask);
```

Comment:
This can't be right. `successfulTask` was already declared to be successful, so this isn't going to do anything. I know I said before that it's important to wait on a task to know the test is ready to proceed, but this task is created right here in this method, and there's no work that it waits on, so this really isn't doing anything. So, if there's something that needs to happen before the test continues, make sure to wait on that, but this seems like a no-op as it is.

Reply:
Cool, I actually wasn't sure about this point, so I just cargo-culted it from my other tests, but  I did think it was a bit odd. I'll remove it. :)